src/libraries/System.Runtime.Numerics/tests/BigInteger/op_rightshift.cs

```diff
@@ -25,6 +25,19 @@ public static void BigShiftsTest()
             }
         }
 
+        [Fact]
+        public static void LargeNegativeBigIntegerShiftTest()
+        {
+            // Create a very large negative BigInteger
+            BigInteger bigInt = new BigInteger(-1) << int.MaxValue;
+            Assert.Equal(2147483647, bigInt.GetBitLength());
+
+            // Right shift the BigInteger and ensure the operation behaves as expected
+            BigInteger shiftedBigInt = bigInt >> 1;
+            Assert.Equal(2147483646, shiftedBigInt.GetBitLength());
+            Assert.True(shiftedBigInt != new BigInteger(-1));
```

Comment:
Should we verify the new value? How do we know it's not all zero..

Reply:
Would be great to validate the result is equal as well here (might be worth a quick check that any other tests are also validating result and not just "shape"). Otherwise LGTM.

Reply:
Good suggestion, taking a look at this now.